test/specs/actions/faye.spec.js

```diff
@@ -342,11 +342,76 @@ describe('Faye Actions', () => {
     });
 
     describe('subscribe', () => {
-        it('should call setFayeSubcription', () => {
-            mockedStore = createMockedStore(sandbox, generateBaseStoreProps());
-            mockedStore.dispatch(fayeActions.subscribe()).then(() => {
-                setFayeSubscriptionSpy.should.have.been.calledOnce;
+        let getClientStub;
+        let subscribeStub;
+
+        beforeEach(() => {
+            subscribeStub = sandbox.stub().resolves();
+
+            getClientStub = sandbox.stub().returnsSyncThunk({
```

Comment:
I think if we didn't have `returnsSyncThunk` this code could be:

```
cosnt client = {
    subscribe: subscribeStub
}
getClientStub = sandbox.stub().returns(() => client);
```

which is a lot clearer if you ask me /2cents